src/ResourceManagement/Sql/SqlFailoverGroupImpl.cs

```diff
@@ -0,0 +1,430 @@
+        ///GENMHASH:10321C7CB3A1E7C461BBEBEAA7FCEB2A:D04715006C36394109746FEBD7928CCE
+        internal SqlFailoverGroupImpl(string name, SqlServerImpl parent, FailoverGroupInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, parent)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = parent.ResourceGroupName;
+            this.sqlServerName = parent.Name;
+            this.sqlServerLocation = parent.RegionName;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name.</param>
+        /// <param name="sqlServerName">The parent SQL server name.</param>
+        /// <param name="name">The name of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses failover group operations.</param>
+        ///GENMHASH:2FFBF1911C3D11A4A0D6DB160D9678D8:A6FEE4EFEDACA027A0C59D9323BFCAC7
+        internal SqlFailoverGroupImpl(string resourceGroupName, string sqlServerName, string sqlServerLocation, string name, FailoverGroupInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+            this.sqlServerLocation = sqlServerLocation;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="name">The name of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses failover group operations.</param>
+        ///GENMHASH:FFACBC94C1D9864074BFA5694BF5256F:8A4C506BC6C1E988E601BB237AB86255
+        internal SqlFailoverGroupImpl(string name, FailoverGroupInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            if (innerObject != null && innerObject.Id != null)
+            {
+                try
+                {
+                    ResourceId resourceId = ResourceId.FromString(innerObject.Id);
+                    this.resourceGroupName = resourceId.ResourceGroupName;
+                    this.sqlServerName = resourceId.Parent.Name;
+                    this.sqlServerLocation = innerObject.Location;
+                }
+                catch (Exception) { }
```

Comment:
what is the point of catching all in this simple case?

Reply:
If the inner freshly created, it does not have a valid ID.